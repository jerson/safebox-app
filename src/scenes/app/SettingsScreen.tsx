import React from 'react';
import { StyleSheet, ViewStyle, ScrollView, StatusBar } from 'react-native';
import { SafeAreaView } from 'react-navigation';
import Colors from '../../modules/constants/Colors';
import Container from '../../components/ui/Container';
import Content from '../../components/ui/Content';
import Size from '../../modules/dimensions/Size';
import DeviceSettings from '../../components/settings/DeviceSettings';
import SessionSettings from '../../components/settings/SessionSettings';
import PremiumSettings from '../../components/settings/PremiumSettings';

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.grey1
  } as ViewStyle,
  scrollView: {
    flex: 1
  } as ViewStyle,
  safeArea: {
    flex: 1
  } as ViewStyle,
  block: {
    marginBottom: 40
  } as ViewStyle
});

function SettingsScreen() {
  return (
    <Container style={styles.container}>
      <StatusBar
        animated
        barStyle={'dark-content'}
        backgroundColor={Colors.grey2}
      />
      <ScrollView
        keyboardShouldPersistTaps={'handled'}
        contentContainerStyle={{
          minHeight: Size.getVisibleTabScreenHeight()
        }}
        style={styles.scrollView}
      >
        <SafeAreaView style={styles.safeArea}>
          <Content>
            <DeviceSettings style={styles.block} />
            <PremiumSettings style={styles.block} />
            <SessionSettings style={styles.block} />
          </Content>
        </SafeAreaView>
      </ScrollView>
    </Container>
  );
}

SettingsScreen.navigationOptions = {
  title: 'Settings',
  headerLeft: null
};

export default SettingsScreen;
