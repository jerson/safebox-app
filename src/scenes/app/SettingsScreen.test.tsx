import React from 'react';
import * as renderer from 'react-test-renderer';
import SettingsScreen from './SettingsScreen';

const props: SettingsScreen = {};
test('renders without crashing', () => {
  const snapshot = renderer.create(<SettingsScreen {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<SettingsScreen {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
