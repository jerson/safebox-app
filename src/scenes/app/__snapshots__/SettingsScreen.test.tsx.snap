// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  style={
    Array [
      Object {
        "backgroundColor": "#fafdff",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#fafdff",
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "minHeight": 1220,
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <_class>
      <_class
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 20,
          }
        }
      >
        <_class
          style={
            Array [
              Object {
                "paddingHorizontal": 20,
              },
              false,
              Object {
                "flex": 1,
                "paddingTop": 20,
              },
              undefined,
            ]
          }
        >
          <_class
            style={
              Array [
                Object {
                  "backgroundColor": "#fafdff",
                },
                Object {
                  "marginBottom": 40,
                },
              ]
            }
          >
            <_class
              style={
                Array [
                  Object {
                    "alignItems": "center",
                  },
                  Object {
                    "marginBottom": 5,
                  },
                ]
              }
            >
              <_class
                style={
                  Array [
                    Object {
                      "backgroundColor": "#9066f5",
                      "height": 1,
                      "left": 0,
                      "opacity": 0.5,
                      "position": "absolute",
                      "right": 0,
                      "top": 12,
                    },
                    Object {
                      "backgroundColor": "#c2d0e7",
                    },
                    false,
                  ]
                }
              />
              <_class
                style={
                  Array [
                    Object {
                      "backgroundColor": "#4531cd",
                      "borderRadius": 10,
                      "paddingBottom": 4,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 6,
                    },
                    Object {
                      "backgroundColor": "#6781a6",
                    },
                    false,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Object {},
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 13,
                        "lineHeight": 14,
                        "minWidth": 20,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Biometric settings
                </Text>
              </_class>
            </_class>
            <_class
              style={
                Object {
                  "alignItems": "center",
                  "marginTop": 5,
                }
              }
            >
              <_class
                style={
                  Object {
                    "marginBottom": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Object {},
                      Object {
                        "color": "#65748b",
                        "fontSize": 12,
                      },
                    ]
                  }
                >
                  you can add additional authentication methods such as fingerprint or facial validation, depending on your device.
                </Text>
              </_class>
              <_class
                style={
                  Object {
                    "backgroundColor": "#f2f4f9",
                    "borderRadius": 10,
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Object {},
                      Object {
                        "color": "#38414e",
                        "fontSize": 13,
                      },
                    ]
                  }
                >
                  your device does not have any supported method, try another device.
                </Text>
              </_class>
            </_class>
            <_class />
          </_class>
          <_class
            style={
              Array [
                Object {
                  "backgroundColor": "#fafdff",
                },
                Object {
                  "marginBottom": 40,
                },
              ]
            }
          >
            <_class
              style={
                Array [
                  Object {
                    "alignItems": "center",
                  },
                  undefined,
                ]
              }
            >
              <_class
                style={
                  Array [
                    Object {
                      "backgroundColor": "#9066f5",
                      "height": 1,
                      "left": 0,
                      "opacity": 0.5,
                      "position": "absolute",
                      "right": 0,
                      "top": 12,
                    },
                    Object {
                      "backgroundColor": "#c2d0e7",
                    },
                    false,
                  ]
                }
              />
              <_class
                style={
                  Array [
                    Object {
                      "backgroundColor": "#4531cd",
                      "borderRadius": 10,
                      "paddingBottom": 4,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 6,
                    },
                    Object {
                      "backgroundColor": "#6781a6",
                    },
                    false,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Object {},
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 13,
                        "lineHeight": 14,
                        "minWidth": 20,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Session settings
                </Text>
              </_class>
            </_class>
            <_class
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <_class
                style={
                  Object {
                    "flex": 1,
                    "paddingRight": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Object {},
                      Object {
                        "color": "#65748b",
                        "fontSize": 12,
                      },
                    ]
                  }
                >
                  This option allows you to safely exit the application.
                </Text>
              </_class>
              <_class
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#6781a6",
                    "borderColor": "#6781a6",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "margin": 0,
                    "minWidth": 120,
                    "opacity": 1,
                    "padding": 12,
                    "zIndex": 1,
                  }
                }
                testID="button_touchable"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Object {},
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 14,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#FFFFFF",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Sign out
                </Text>
              </_class>
            </_class>
          </_class>
        </_class>
      </_class>
    </_class>
  </RCTScrollView>
</_class>
`;
